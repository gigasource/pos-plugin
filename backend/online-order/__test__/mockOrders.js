const { ObjectID } = require('bson')

module.export = [{
	"printer": [],
	"_id": new ObjectID("5ffffadd207bc21ece71a2db"),
	"id": 1,
	"status": "completed",
	"items": [{
		"taxes": [],
		"_id": new ObjectID("5ffffadd207bc21ece71a2dc"),
		"id": "",
		"name": "Pepe",
		"price": 12,
		"tax": 7,
		"quantity": 1,
		"modifiers": [],
		"groupPrinter": "Kitchen",
		"category": "1",
		"originalPrice": 12,
		"attributes": []
	}],
	"customer": {
		"name": "fgh",
		"company": "",
		"phone": "0123123131",
		"address": "",
		"zipCode": "",
		"deliveryTime": "",
		"distance": null
	},
	"deliveryDate": new Date("2021-01-14T08:03:41.193Z"),
	"payment": [{"_id": new ObjectID("5e0b00e1c0aa5a17f5ab84ab"), "value": 12, "type": "cash"}],
	"type": "pickup",
	"date": new Date("2021-01-14T08:03:36.457Z"),
	"vDate": new Date("2021-01-13T17:00:00.000Z"),
	"bookingNumber": "210114150336457",
	"shippingFee": 0,
	"vSum": 12,
	"vTax": 0.79,
	"vTaxGroups": [{"_id": new ObjectID("5ffffadd207bc21ece71a2de"), "taxType": 7, "tax": 0.79, "sum": 12}],
	"vDiscount": 0,
	"online": true,
	"note": "",
	"onlineOrderId": "0159c1c0-563f-11eb-accb-3564ab607d04",
	"discounts": [],
	"deliveryTime": "15:33",
	"user": [{"_id": new ObjectID("5e1d89f200971c2ac29fcccb"), "name": "admin"}],
	"commits": [],
	"declineReason": "",
	"prepareTime": 30
}, {
	"printer": [],
	"_id": new ObjectID("600fe75dbd5c7e5549974874"),
	"id": 2,
	"status": "kitchen",
	"items": [{
		"taxes": [],
		"_id": new ObjectID("600fe75dbd5c7e5549974875"),
		"id": "3",
		"image": "/cms-files/files/view/Restaurant%20Plus%20images/75ABE9D7-3244-494D-BDF3-F6E3EED92699.png",
		"name": "mon test",
		"desc": "In publishing and graphic design, Lorem ipsum is a placeholder text commonly used to demonstrate the visual form of a document or a typeface without relying on meaningful content. ",
		"price": 0.1,
		"tax": 7,
		"quantity": 1,
		"modifiers": [],
		"groupPrinter": "Kitchen",
		"groupPrinter2": "false",
		"category": "1",
		"originalPrice": 0.1,
		"attributes": []
	}],
	"customer": {
		"name": "xa",
		"company": "",
		"phone": "01234",
		"address": "",
		"zipCode": "",
		"deliveryTime": "",
		"distance": null
	},
	"deliveryDate": new Date("2021-01-26T09:56:45.029Z"),
	"payment": [{"_id": new ObjectID("5e0b00e1c0aa5a17f5ab84ab"), "value": 0.1, "type": "cash"}],
	"type": "pickup",
	"date": new Date("2021-01-26T09:56:43.201Z"),
	"timeoutDate": new Date("2021-01-26T10:01:43.201Z"),
	"vDate": new Date("2021-01-25T17:00:00.000Z"),
	"bookingNumber": "210126165643201",
	"shippingFee": 0,
	"vSum": 0.1,
	"vTax": 0.01,
	"vTaxGroups": [{"_id": new ObjectID("600fe75dbd5c7e5549974877"), "taxType": 7, "tax": 0.01, "sum": 0.1}],
	"vDiscount": 0,
	"online": true,
	"note": "",
	"onlineOrderId": "cb60ef30-5fbc-11eb-a77a-a1ad16df7dd7",
	"discounts": [],
	"deliveryTime": "17:27",
	"user": [{"_id": new ObjectID("5e1d89f200971c2ac29fcccb"), "name": "admin"}],
	"commits": [],
	"declineReason": "",
	"prepareTime": 30
}, {
	"printer": [],
	"_id": new ObjectID("600fe788bd5c7e554997488f"),
	"id": 3,
	"status": "kitchen",
	"items": [{
		"taxes": [],
		"_id": new ObjectID("600fe788bd5c7e5549974890"),
		"id": "4",
		"image": "/cms-files/files/view/images/DSCF7485.jpg",
		"name": "VIT THAI XOAI | MANGO IN THE â€žDUCK\"",
		"price": 9,
		"tax": 7,
		"quantity": 1,
		"modifiers": [],
		"groupPrinter": "Kitchen",
		"groupPrinter2": "false",
		"category": "1",
		"originalPrice": 10,
		"discountUnit": "amount",
		"vDiscount": 1,
		"attributes": []
	}],
	"customer": {
		"name": "xa",
		"company": "",
		"phone": "0123",
		"address": "",
		"zipCode": "",
		"deliveryTime": "",
		"distance": null
	},
	"deliveryDate": new Date("2021-01-26T09:57:28.770Z"),
	"payment": [{"_id": new ObjectID("5e0b00e1c0aa5a17f5ab84ab"), "value": 9, "type": "cash"}],
	"type": "pickup",
	"date": new Date("2021-01-26T09:57:26.930Z"),
	"timeoutDate": new Date("2021-01-26T10:02:26.930Z"),
	"vDate": new Date("2021-01-25T17:00:00.000Z"),
	"bookingNumber": "210126165726930",
	"shippingFee": 0,
	"vSum": 9,
	"vTax": 0.59,
	"vTaxGroups": [{"_id": new ObjectID("600fe788bd5c7e5549974892"), "taxType": 7, "tax": 0.59, "sum": 9}],
	"vDiscount": 1,
	"online": true,
	"note": "",
	"onlineOrderId": "e571c160-5fbc-11eb-a77a-a1ad16df7dd7",
	"discounts": [{"_id": new ObjectID("600fe788bd5c7e5549974893"), "name": "Giam gia test", "value": 1, "type": "percent"}],
	"deliveryTime": "17:27",
	"user": [{"_id": new ObjectID("5e1d89f200971c2ac29fcccb"), "name": "admin"}],
	"commits": [],
	"declineReason": "",
	"prepareTime": 30
}]
