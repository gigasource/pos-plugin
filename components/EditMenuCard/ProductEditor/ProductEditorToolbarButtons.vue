<script>
import constants from '../EditMenuCardToolbar/constants';
import { getCurrentInstance, ref } from 'vue';
import { useI18n } from 'vue-i18n'
import { genScopeId } from '../../utils';
import { ProductEditModes, updateProductEditMode, view } from '../../OrderView/pos-ui-shared';
import { canCopy, canDelete, canSwitch, deleteProductLayout, setAction } from './ProductEditorLogic';

const shouldShowSwitchEditModeDialog = ref(true)
const showSwitchEditModeDialog = ref(false)
const openSwitchModeDialogIfNeeded = () => {
  if (shouldShowSwitchEditModeDialog.value) {
    showSwitchEditModeDialog.value = true
    shouldShowSwitchEditModeDialog.value = false
  }
}

export default {
  name: 'ProductEditorToolbarButtons',
  props: {},
  setup() {
    const { t } = useI18n()
    const changeToIngredientMode = () => {
      openSwitchModeDialogIfNeeded()
      updateProductEditMode(ProductEditModes.ingredient)
    }

    // change position btn
    const renderSwitchBtn = () =>
        <g-btn-bs elevation="2" icon="icon-edit-menu-card-switch"
                  onClick={() => setAction('switch')}
                  disabled={!canSwitch.value}>{t('ui.switch')}</g-btn-bs>

    // clone button
    const renderCopyBtn = () =>
        <g-btn-bs elevation="2" icon="icon-edit-menu-card-copy"
                  onClick={() => setAction('copy')}
                  disabled={!canCopy.value}>{t('ui.copy')}</g-btn-bs>

    // delete button
    const showDeleteConfirmDialog = ref(false)
    const renderDeleteBtn = () => <>
      <g-btn-bs elevation="2" icon="icon-edit-menu-card-delete" onClick={() => showDeleteConfirmDialog.value = true} disabled={!canDelete.value}>{t('ui.delete')}</g-btn-bs>
      <dialog-confirm-delete v-model={showDeleteConfirmDialog.value} type=' this product' onSubmit={() => {
        deleteProductLayout();
        showDeleteConfirmDialog.value = false;
      }}></dialog-confirm-delete>
    </>

    // switch edit mode button
    const renderSwitchEditModeBtn = () => (view.value.mode === ProductEditModes.ingredient
            ? <g-btn-bs elevation="2" icon="icon-ingredient-mode"
                        onClick={() => updateProductEditMode(ProductEditModes.basic)}>{t('inventory.ingredientMode')} </g-btn-bs>
            : <g-btn-bs elevation="2" icon="icon-basic-mode"
                        onClick={changeToIngredientMode}>{t('inventory.basicMode')} </g-btn-bs>
    )
    const renderSwitchEditModeDialog = () => (
        <g-dialog v-model={showSwitchEditModeDialog.value} eager width="448">
          {genScopeId(() => (
              <div class="dialog" onClick={() => showSwitchEditModeDialog.value = false}>
                <div class="dialog-content">
                  <g-icon>icon-basic-mode</g-icon>
                  <div style="flex: 1; margin-left: 16px">
                    <p class="dialog-content__title">{t('inventory.basicMode')}</p>
                    <p class="dialog-content__detail">{t('inventory.basicNote')}</p>
                  </div>
                </div>
                <div class="dialog-content">
                  <g-icon>icon-ingredient-mode</g-icon>
                  <div style="flex: 1; margin-left: 16px">
                    <p class="dialog-content__title">{t('inventory.ingredientMode')}</p>
                    <p class="dialog-content__detail">{t('inventory.ingredientNote')}</p>
                  </div>
                </div>
                <div class="dialog-message">{t('inventory.clickDismiss')}</div>
              </div>
          ))()}
        </g-dialog>
    )

    return () => <>
        <portal to={constants.portalLeftButtons}>
          {
            genScopeId(() => [
              renderSwitchBtn(),
              renderDeleteBtn(),
              renderCopyBtn()
            ])()
          }
        </portal>
        <portal to={constants.portalRightButtons}>
          {
            genScopeId(() => [
              renderSwitchEditModeBtn(),
              renderSwitchEditModeDialog()
            ])()
          }
        </portal>
    </>
  }
}
</script>
<style scoped lang="scss">
@import "../EditMenuCardToolbar/emc-button.scss";

.dialog {
  width: 100%;
  background-color: white;
  border-radius: 4px;
  padding: 20px;

  &-content {
    display: flex;
    margin-bottom: 16px;

    &__title {
      font-size: 14px;
      font-weight: 700;
      margin-bottom: 4px;
    }

    &__detail {
      font-size: 13px;
    }
  }

  &-message {
    text-align: center;
    font-size: 12px;
    color: #9e9e9e;
  }
}

</style>
