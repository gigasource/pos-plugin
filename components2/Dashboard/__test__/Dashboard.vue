<script>
import ContentRender from "../../../components/common/ContentRender";
import {KeepAlive, onActivated, ref, h} from "vue";
import {contentView} from "./dashboard-shared";
import _ from 'lodash';


const TextField = {
  setup(props) {
    const count = ref(0);
    const init = _.debounce(() => {
      count.value ++;
    }, 10);
    init();
    onActivated(() => {
      init();
      console.log('onActivated');
    })

    return () => <p>{count.value}</p>
  }
}

const TextComp = function () {
  return <p>c</p>
}

const views = {
  TextComp,
  TextField
}

export default {
  name: "Dashboard",
  components: {
    ContentRender,
  },
  setup() {
    return () => <KeepAlive>
      {h(views[contentView.value.name])}
    </KeepAlive>
  }
}
</script>

<style scoped>

</style>
