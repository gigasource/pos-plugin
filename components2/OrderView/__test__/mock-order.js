const mockOrder = {
  table: "10",
  _id: "5fee6600612d45b5ffdcabb7",
  items:
    [{
      groupPrinter: "Bar",
      id: "48",
      name: "Whiskey",
      price: 40,
      product: "5e784d7786e3ec0997b5e39b",
      taxes: [19, 7],
      course: 1,
      quantity: 1,
      _id: "5fee6600612d45b5ffdcabb8",
      vSum: 40,
      vTakeAway: false,
      tax: 19,
      sent: true,
      printed: true,
      vTaxSum: {
        "19": {
          tax: 6.39,
          net: 33.61,
          gross: 40
        }
      }
    }],
  cancellationItems: [],
  takeAway: false,
  status: "inProgress",
  payment: [],
  user: [],
  vSum: 40,
  vDate: "2020-12-31T17:00:00.000Z",
  vTaxSum: {
    '19': {
      tax: 6.39,
      net: 33.61,
      gross: 40
    }
  }
}

module.exports = {
  mockOrder
}
