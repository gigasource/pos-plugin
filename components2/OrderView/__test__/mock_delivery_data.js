import { mockProducts } from './mock_products'
import _ from 'lodash'

export const mockFavorites = _.filter(mockProducts, p => p.option && p.option.favorite)
export const selectedCustomerMock = {
  'name': 'Tony',
  'phone': '0988.888.888',
  'addresses': [
    {
      address: 'address val 1',
      zipcode: 'zipcode val 1',
      house: 'hose val 1',
      street: 'street val 1',
      city: 'city val 1'
    },
    {
      address: 'address val 2',
      zipcode: 'zipcode val 2',
      house: 'hose val 2',
      street: 'street val 2',
      city: 'city val 2'
    }
  ],
}

export default {
  'favorites': mockFavorites, // favorite products
  //
  'selectedCustomer': selectedCustomerMock,

  // index of address will be used to ship product (in case the selected customer have more than 1 addrs)
  // TODO: [consider] rename to selectedAddressIndex
  'selectedAddress': 0,

  'address': '', // show in tablet mode for new customer
  'dialogMode': 'add', // edit
  'autocompleteAddresses': [], // only show in mobile mode (gather by query to cms backend)
  'placeId': '', // show in dialog input slot & deliveryOrderMode='mobile'
  'showKeyboard': false, // true
  'deliveryOrderMode': 'tablet', // mobile

  //
  'calls': [],
  'missedCalls': [],

  // doesn't show in snapshot (using in non-default slots)
  'name': 'Tommy',
  'phone': '084.999.8888',
  'zipcode': '',
  'city': '',
  'house': '',
  'street': '',
}
