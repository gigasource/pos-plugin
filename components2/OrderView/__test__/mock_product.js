export const mockProduct = {
  "_id": "5e784d7786e3ec0997b5e39b",
  "category": {
    "_id": "5df6f5dd7854273fa4fd989e",
    "name": "Wine",
    "position": 5
  },
  "groupPrinter": {
    "_id": "5e7864b186e3ec0997b5e48e",
    "name": "Bar",
    "printer": "HP M176n",
    "printers": [
      {
        "_id": "5e81774685fb2d03dda95cd0",
        "printerType": "ip",
        "ip": "192.168.10.6",
        "escPOS": false,
        "merge": "true"
      }
    ],
    "showDineInTax": false,
    "type": "kitchen"
  },
  "id": "48",
  "layouts": [
    {
      "_id": "5e784d7786e3ec0997b5e39c",
      "color": "#DCE775",
      "order": 2,
      "device": "pos"
    }
  ],
  "name": "Whiskey",
  "option": {},
  "price": 40,
  "tax": 19,
  "tax2": 7
}
