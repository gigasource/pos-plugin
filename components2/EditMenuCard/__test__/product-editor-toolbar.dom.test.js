import {
  wrapper,
  makeWrapper,
} from "../../test-utils";
const delay = require("delay");
import {nextTick} from 'vue'
import {
  orderLayout,
  ProductEditModes,
  selectCategoryLayout,
  selectProductLayout,
  view
} from '../../OrderView/pos-ui-shared';
import _ from 'lodash';
import { mockProduct } from '../../OrderView/__test__/mock_product';
const ignoreComponents = ['Portal', 'PortalTarget', 'MountingPortal', 'portal', 'portal-target']
import ProductEditorToolbarButtons from '../ProductEditor/ProductEditorToolbarButtons';

describe('product-editor-toolbar', () => {
  function setupProduct(product, productLayout = {}) {
    // setup order layout
    orderLayout.value = { rows: 1, columns: 3, type: 'default', categories: [
        {
          row: 1, column: 2, top: 0, left: 0,
          products: [
            { product, top: 0, left: 0, ...productLayout}
          ]
        }
      ]}
    selectCategoryLayout({ top: 0, left: 0 })
    selectProductLayout({ top: 0, left: 0 })
  }

  async function expectSnapshot() {
    ProductEditorToolbarButtons.components = {}
    makeWrapper(ProductEditorToolbarButtons, {
      shallow: false,
      global: {
        stubs: {
          portal: true,
          'dialog-confirm-delete': true,
          'color-selector': true,
          'input-number': true,
          'dialog-text-filter': true,
          'g-select': true,
          'g-switch': true,
          'dialog-product-info': true,
          'dialog-edit-popup-modifiers': true
        },
        config: {
          isCustomElement: tag => _.includes(ignoreComponents, tag)
        }
      }
    }, false)

    await nextTick()
    await delay(50)
    expect(wrapper.html()).toMatchSnapshot()
  }

  it('should render, allow switch, copy, delete when product is real', async () => {
    setupProduct(mockProduct, { _id: 'prodLayoutId' })
    await expectSnapshot()
  })

  it('should render, not allow to switch, copy, delete when product is empty', async () => {
    setupProduct({...mockProduct})
    await expectSnapshot()
  })

  it('should render switch to ingredient mode when current mode is basic', async () => {
    view.value.mode = ProductEditModes.basic
    setupProduct({...mockProduct})
    await expectSnapshot()
  })

  it('should render switch to basic mode when current mode is ingredient', async () => {
    view.value.mode = ProductEditModes.ingredient
    setupProduct({...mockProduct})
    await expectSnapshot()
  })
});
