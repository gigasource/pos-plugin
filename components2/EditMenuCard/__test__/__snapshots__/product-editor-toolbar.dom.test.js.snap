// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`product-editor-toolbar should render switch to basic mode when current mode is ingredient 1`] = `
<portal-stub to="portal-left-buttons">
  <g-btn-bs-stub elevation="2"
                 icon="icon-edit-menu-card-switch"
                 disabled="true"
  >
    ui.switch
  </g-btn-bs-stub>
  <g-btn-bs-stub elevation="2"
                 icon="icon-edit-menu-card-delete"
                 disabled="true"
  >
    ui.delete
  </g-btn-bs-stub>
  <dialog-confirm-delete-stub modelvalue="false"
                              type=" this product"
  >
  </dialog-confirm-delete-stub>
  <g-btn-bs-stub elevation="2"
                 icon="icon-edit-menu-card-copy"
                 disabled="true"
  >
    ui.copy
  </g-btn-bs-stub>
</portal-stub>
<portal-stub to="portal-right-buttons">
  <g-btn-bs-stub elevation="2"
                 icon="icon-ingredient-mode"
  >
    inventory.ingredientMode
  </g-btn-bs-stub>
  <g-dialog-stub modelvalue="false"
                 eager="true"
                 width="448"
  >
    <div class="dialog">
      <div class="dialog-content">
        <g-icon-stub>
          icon-basic-mode
        </g-icon-stub>
        <div style="flex: 1; margin-left: 16px;">
          <p class="dialog-content__title">
            inventory.basicMode
          </p>
          <p class="dialog-content__detail">
            inventory.basicNote
          </p>
        </div>
      </div>
      <div class="dialog-content">
        <g-icon-stub>
          icon-ingredient-mode
        </g-icon-stub>
        <div style="flex: 1; margin-left: 16px;">
          <p class="dialog-content__title">
            inventory.ingredientMode
          </p>
          <p class="dialog-content__detail">
            inventory.ingredientNote
          </p>
        </div>
      </div>
      <div class="dialog-message">
        inventory.clickDismiss
      </div>
    </div>
  </g-dialog-stub>
</portal-stub>
`;

exports[`product-editor-toolbar should render switch to ingredient mode when current mode is basic 1`] = `
<portal-stub to="portal-left-buttons">
  <g-btn-bs-stub elevation="2"
                 icon="icon-edit-menu-card-switch"
                 disabled="true"
  >
    ui.switch
  </g-btn-bs-stub>
  <g-btn-bs-stub elevation="2"
                 icon="icon-edit-menu-card-delete"
                 disabled="true"
  >
    ui.delete
  </g-btn-bs-stub>
  <dialog-confirm-delete-stub modelvalue="false"
                              type=" this product"
  >
  </dialog-confirm-delete-stub>
  <g-btn-bs-stub elevation="2"
                 icon="icon-edit-menu-card-copy"
                 disabled="true"
  >
    ui.copy
  </g-btn-bs-stub>
</portal-stub>
<portal-stub to="portal-right-buttons">
  <g-btn-bs-stub elevation="2"
                 icon="icon-basic-mode"
  >
    inventory.basicMode
  </g-btn-bs-stub>
  <g-dialog-stub modelvalue="false"
                 eager="true"
                 width="448"
  >
    <div class="dialog">
      <div class="dialog-content">
        <g-icon-stub>
          icon-basic-mode
        </g-icon-stub>
        <div style="flex: 1; margin-left: 16px;">
          <p class="dialog-content__title">
            inventory.basicMode
          </p>
          <p class="dialog-content__detail">
            inventory.basicNote
          </p>
        </div>
      </div>
      <div class="dialog-content">
        <g-icon-stub>
          icon-ingredient-mode
        </g-icon-stub>
        <div style="flex: 1; margin-left: 16px;">
          <p class="dialog-content__title">
            inventory.ingredientMode
          </p>
          <p class="dialog-content__detail">
            inventory.ingredientNote
          </p>
        </div>
      </div>
      <div class="dialog-message">
        inventory.clickDismiss
      </div>
    </div>
  </g-dialog-stub>
</portal-stub>
`;

exports[`product-editor-toolbar should render, allow switch, copy, delete when product is real 1`] = `
<portal-stub to="portal-left-buttons">
  <g-btn-bs-stub elevation="2"
                 icon="icon-edit-menu-card-switch"
                 disabled="false"
  >
    ui.switch
  </g-btn-bs-stub>
  <g-btn-bs-stub elevation="2"
                 icon="icon-edit-menu-card-delete"
                 disabled="false"
  >
    ui.delete
  </g-btn-bs-stub>
  <dialog-confirm-delete-stub modelvalue="false"
                              type=" this product"
  >
  </dialog-confirm-delete-stub>
  <g-btn-bs-stub elevation="2"
                 icon="icon-edit-menu-card-copy"
                 disabled="false"
  >
    ui.copy
  </g-btn-bs-stub>
</portal-stub>
<portal-stub to="portal-right-buttons">
  <g-btn-bs-stub elevation="2"
                 icon="icon-basic-mode"
  >
    inventory.basicMode
  </g-btn-bs-stub>
  <g-dialog-stub modelvalue="false"
                 eager="true"
                 width="448"
  >
    <div class="dialog">
      <div class="dialog-content">
        <g-icon-stub>
          icon-basic-mode
        </g-icon-stub>
        <div style="flex: 1; margin-left: 16px;">
          <p class="dialog-content__title">
            inventory.basicMode
          </p>
          <p class="dialog-content__detail">
            inventory.basicNote
          </p>
        </div>
      </div>
      <div class="dialog-content">
        <g-icon-stub>
          icon-ingredient-mode
        </g-icon-stub>
        <div style="flex: 1; margin-left: 16px;">
          <p class="dialog-content__title">
            inventory.ingredientMode
          </p>
          <p class="dialog-content__detail">
            inventory.ingredientNote
          </p>
        </div>
      </div>
      <div class="dialog-message">
        inventory.clickDismiss
      </div>
    </div>
  </g-dialog-stub>
</portal-stub>
`;

exports[`product-editor-toolbar should render, not allow to switch, copy, delete when product is empty 1`] = `
<portal-stub to="portal-left-buttons">
  <g-btn-bs-stub elevation="2"
                 icon="icon-edit-menu-card-switch"
                 disabled="true"
  >
    ui.switch
  </g-btn-bs-stub>
  <g-btn-bs-stub elevation="2"
                 icon="icon-edit-menu-card-delete"
                 disabled="true"
  >
    ui.delete
  </g-btn-bs-stub>
  <dialog-confirm-delete-stub modelvalue="false"
                              type=" this product"
  >
  </dialog-confirm-delete-stub>
  <g-btn-bs-stub elevation="2"
                 icon="icon-edit-menu-card-copy"
                 disabled="true"
  >
    ui.copy
  </g-btn-bs-stub>
</portal-stub>
<portal-stub to="portal-right-buttons">
  <g-btn-bs-stub elevation="2"
                 icon="icon-basic-mode"
  >
    inventory.basicMode
  </g-btn-bs-stub>
  <g-dialog-stub modelvalue="false"
                 eager="true"
                 width="448"
  >
    <div class="dialog">
      <div class="dialog-content">
        <g-icon-stub>
          icon-basic-mode
        </g-icon-stub>
        <div style="flex: 1; margin-left: 16px;">
          <p class="dialog-content__title">
            inventory.basicMode
          </p>
          <p class="dialog-content__detail">
            inventory.basicNote
          </p>
        </div>
      </div>
      <div class="dialog-content">
        <g-icon-stub>
          icon-ingredient-mode
        </g-icon-stub>
        <div style="flex: 1; margin-left: 16px;">
          <p class="dialog-content__title">
            inventory.ingredientMode
          </p>
          <p class="dialog-content__detail">
            inventory.ingredientNote
          </p>
        </div>
      </div>
      <div class="dialog-message">
        inventory.clickDismiss
      </div>
    </div>
  </g-dialog-stub>
</portal-stub>
`;
